pkg/controller/volume/persistentvolume/volume_host.go

```diff
@@ -136,3 +138,13 @@ func (ctrl *PersistentVolumeController) GetSubpather() subpath.Interface {
 	// No volume plugin needs Subpaths in PV controller.
 	return nil
 }
+
+func (ctrl *PersistentVolumeController) GetCSIDriverLister() storagelisters.CSIDriverLister {
```

Comment:
you can use the new `AttachDetachVolumeHost` interface instead to avoid adding this interface to all the other controllers
https://github.com/kubernetes/kubernetes/blob/83ff0f6c64485ffe491d5116596072f466be1bd5/pkg/volume/plugins.go#L326

Reply:
I tried this but couldn't find a clean way to pass VolumeHost through InitPlugins (plugins.go) and then to plugin.Init(host).  Either I need to pass both VolumeHost and AttachDetachVolumeHost or pass a host object with the combination of them.  Let me know if you have any suggestions.

Reply:
By removing the need for CSIDriverLister in csi.Init(), then I don't think you have this issue anymore, and can have the two separate volume host interfaces.

Reply:
I think I would still have the same issue.  VolumeHost is passed in plugin.Init(host).  Even though we don't need CSIDriverLister in init(), it is already passed in as part of host so we can use it in other places when needed.  If I use AttachDetachVolumeHost, I would still need to pass in both VolumeHost and AttachDetachVolumeHost thru Init.

Reply:
@xing-yang you should be able to just pass in the `VolumeHost` and you can do a type assertion on `AttachDetachvolumeHost` which will succeed if it is the `attachDetachController`.

You can look at this as an example: https://github.com/kubernetes/kubernetes/pull/74835/files#diff-450e811a4953f760ff1594ede8b2037eR1589

Reply:
I also need that for kubelet, not just the AD controller.  If I add CSIDriverLister in both KubeletVolumeHost and AttachDetachVolumeHost, that seems redundant and keeping it in VolumeHost seems better.

Reply:
The long term plan is to no longer have a VolumeHost and just have per-controller interfaces so we only need to pass exactly what we need to each controller.

Reply:
Ok, I'll take a look and see how to pass CSIDriverLister thru both KubeletVolumeHost and AttachDetachVolumeHost.